### PR TITLE
Remove `.babelrc`  in favor of `SWC`

### PR DESCRIPTION
**Pull request by sandylcruz**
# Summary
* We're currently using `next` version `12.1.0`, which allows us to use [`next`'s Rust-based SWC](https://nextjs.org/docs/advanced-features/compiler) instead of our current [.babelrc](https://github.com/gravitational/next/blob/main/.babelrc)
* Because the`/next` repository had a `.babelrc`, we were automatically opted out of using SWC when `next 12` was released (see below message in console when running `yarn dev`)

Screen Shot 2022-05-02 at 9 29 18 AM

* SWC [supports](https://nextjs.org/docs/advanced-features/compiler#styled-components) `styled-components` and `styled-components`'s `ssr`, which were the only things that were in our `.babelrc`
* We can further opt-in to using `SWC` for [minification](https://nextjs.org/docs/advanced-features/compiler#minification). Because this is still considered an experimental feature, it was not added to this PR.

## Why switch to SWC
* `SWC` is up to 17x faster than Babel when compiling individual files and up to 5x faster Fast Refresh. [[source](https://nextjs.org/blog/next-12)]

# Results 

* All tests run 5 times with `yarn build` with mean value provided
* `SWC` build time is **25%** faster than with `.babelrc`
* `SWC + swcMinify` build time is **35%** faster than `.babelrc`
* While `swcMinify` saves 10 seconds of build time, it is still an experimental feature. It should potentially be used once it's a standard feature, and all bugs are worked out.

### Baseline with .babelrc
| Performance | Build time | Build time mean |
| -------------| ----------| -----------------|
| Build time | 88.83s, 84.12s, 79.01s, 79.82s, 78.25s | **82.00s** |

### SWC
| Performance | Build time | Build time mean |
| -------------| ----------| -----------------|
| Build time | 81.94s, 60.83s, 56.21s, 55.84s, 56.14s | **62.19s** |

### SWC + swcMinify
| Performance | Build time | Build time mean |
| -------------| ----------| -----------------|
| Build time | 52.87s, 50.68s, 53.44s, 55.35s, 55.81s | **53.63s** |

# Visible Regressions noticed involving selectors:

| Component | Page / Component example | Regression description |
| ------------| ----- | --------------------------|
| `BenefitCard` | [/about/benefits](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/about/benefits/) |  wrong font size / line height |
| `Code` | [/installing](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/installing/) | wrong font / font-size / syntax highlighting |
| `DropdownMenuItem` | [all pages in navbar after you click 'sign in` ](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/) |  no `mt` between items |
| `Events` layout | [/about/events](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/about/events/) | no hover state on individual events, event image not visible, visited event title link not right color |
| `PromoPointsBlock` |  [451 report](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/resources/analyst-reports/451-report-teleport-modern-pam-solution/) | no `mt` between items |
| `SectionSteps` | [/features/access-requests](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/features/access-requests/) |  no `ml` and `mt` between items |
| `TryTeleport` | [zero trust](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/what-is-zero-trust/) | no visible hover/active/focused state on link |
| `VideoRow` | [Teleport 9 webinar](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/resources/videos/teleport-9-webinar/) | extra `mr` on end video item |
| `WorksWith` |  [/connect](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/connect/) | No dashed boarder on right side / bottom |

# Complex selectors / pseudoselectors, but had no visible change

| Component | Page / Component example | Type |  Rewritten? |
| ------------| ----- | --------------------------| -- |
| `Admonition` |  [/hiw/edge-access](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/how-it-works/ssh-kubernetes-access-on-the-edge/) | Nested CSS |   X |
| `AnchorNavigation` | Not used anywhere | Nested CSS |  X |
| `Column` |  [/connect](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/connect/) | Nested CSS | X |
| `Command` | Not used anywhere |  Pseudo-selector |  |
| `EmailSubscribe` | [2021 security audit](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/resources/audits/teleport-latacora-security-audit-2021/) | Nested CSS | X |
| `IFrame` |  [/resources/videos/teleport-9-webinar](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/resources/videos/teleport-9-webinar/) | Nested CSS | X |
| `Image` |[/ssh-jump-server](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/ssh-jump-server/) | Nested CSS | X |
| `Marketo` | [/legal/subprocessors](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/legal/subprocessors/) |  Pseudo-selector |  |
| `Pre` | [/installing](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/installing/) | Nested CSS | X |
| `Quote` | [/case-study/anaconda](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/case-study/anaconda/) | Pseudo-selector |  |
| `SectionPlan` | [/pricing](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/pricing/) | Nested CSS | X |
| `SectionVideo` | [/features/access-requests](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/features/access-requests/) | Nested CSS | X |
| `SubprocessorSubscribeForm` | [/legal/subprocessors](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/legal/subprocessors/) | Pseudo-selector |  |
| `Tile` | Not used anywhere | Nested CSS | X |
| `TileImage` |  Not used anywhere | Nested CSS | X |
| `TileList` | Not used anywhere | Nested CSS | X |
| `TileSet` | Not used anywhere | Nested CSS | X |
| `/features/access-requests` | [/features/access-requests](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/features/access-requests/) |  Nested CSS | |
| `/installing` | [/installing](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/installing/) | Nested CSS | X |
| `/legal/subprocessors` | [/legal/subprocessors](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/legal/subprocessors/) | Nested CSS | |
| `/pam-for-kubernetes` | [/pam-for-kubernetes](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/pam-for-kubernetes/) | Nested CSS | X |
| `/ssh-jump-server` | [/ssh-jump-server](https://teleport-docs-next-git-sandy-babel-gravitational.vercel.app/ssh-jump-server/) | Nested CSS | |

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202227050660389

### CODE REVIEW

**Comment by iAdramelk** [comment]
@sandylcruz hi sandy. Last time I tried to switch to swc I had to roll back becasuse of large number of regressions (https://github.com/gravitational/next/issues/744). Mostly with complex selectors inside `css={}`.

Did Next.js people fixed them? (Didn't make a detailed review of deployed version yet).

**Comment by sandylcruz** [comment]
@iAdramelk Not sure if it's been fixed, but I'll look into that. If it hasn't, I'll test to see if the complicated selectors can be re-written and simplified. `/blog` is currently using `SWC` by default now though, and I don't think there've been problems.

**Comment by iAdramelk** [comment]
@sandylcruz I think Katya spend few days for testing and fixing broken selectors for the blog to make it work to be able to enable it there. You can ask her for the details about that.

File: next.config.mjs [hunk 1]
```diff
@@ -74,6 +74,11 @@ export default withBundleAnalyzer({
     disableStaticImages: true,
     domains: ["i.ytimg.com"], // Images for youtube preview
   },
+  compiler: {
+    // ssr and displayName are configured by default
+    styledComponents: true,
+  },
+  swcMinify: true,
```
Comment (iAdramelk) [inline next.config.mjs:81]: Forgot to write about it last time. I tried to switch to `swcMinify` then it was just released, but the size of generated files was a lot larger then with default minifier. Was this fixed?

Comment (sandylcruz) [reply next.config.mjs]: The generated file size was similar to the default minifier, but I ended up removing `swcMinify` due to its experimental status.

File: components/Tile/TileImage.tsx [hunk 2]
```diff
@@ -114,3 +76,41 @@ TileImage.defaultProps = {
 };
 
 export default TileImage;
+
+const ButtonLabelLink = styled(Link)(
+  css({
+    border: "1px sold light-gray",
```
Comment (C-STYR) [inline components/Tile/TileImage.tsx:82]:
```suggestion
    border: "1px solid light-gray",
```

File: pages/pam-for-kubernetes/index.mdx [hunk 3]
```diff
@@ -34,19 +34,21 @@ import TryTeleport from "components/TryTeleport";
     />
     <h1>Privileged Access Management (PAM) for Kubernetes</h1>
 
+    <p fontSize="text-xl">
```
Comment (C-STYR) [inline pages/pam-for-kubernetes/index.mdx:37]: this makes me happy - I have a vague memory that directly styling a vanilla html tag didn't use to be possible.

**Review by C-STYR — APPROVED** [review]
Nice work.

**Pull request merged**